modules/nw-sriov-dpdk-example-mellanox.adoc.adoc

```diff
@@ -0,0 +1,146 @@
+[id="example-vf-use-in-dpdk-mode-mellanox_{context}"]
+= Using a virtual function in DPDK mode with a Mellanox NIC
+
+.Prerequisites
+
+* Install the OpenShift CLI (`oc`).
+* Install the SR-IOV Network Operator.
+* Log in as a user with `cluster-admin` privileges.
+
+.Procedure
+
+. Create the following `SriovNetworkNodePolicy` object, and then save the YAML in the `mlx-dpdk-node-policy.yaml` file.
++
+[source,yaml]
+----
+apiVersion: sriovnetwork.openshift.io/v1
+kind: SriovNetworkNodePolicy
+metadata:
+  name: mlx-dpdk-node-policy
+  namespace: openshift-sriov-network-operator
+spec:
+  resourceName: mlxnics
+  nodeSelector:
+    feature.node.kubernetes.io/network-sriov.capable: "true"
+  priority: <priority>
+  numVfs: <num>
+  nicSelector:
+    vendor: "15b3"
+    deviceID: "1015" <1>
+    pfNames: ["<pf_name>", ...]
+    rootDevices: ["<pci_bus_id>", "..."]
+  deviceType: netdevice <2>
+  isRdma: true <3>
+----
+<1> Specify the device hex code of the SR-IOV network device. The only allowed values for Mellanox cards are `1015`, `1017`.
+<2> Specify the driver type for the virtual functions to `netdevice`. Mellanox SR-IOV VF can work in DPDK mode without using the `vfio-pci` device type. VF device appears as a kernel network interface inside a container.
+<3> Enable RDMA mode. This is required for Mellanox cards to work in DPDK mode.
++
+[NOTE]
+=====
+See the `Configuring SR-IOV network devices` section for detailed explanation on each option in the `SriovNetworkNodePolicy` object.
+
+When applying the configuration specified in a `SriovNetworkNodePolicy` object, the SR-IOV Operator may drain the nodes, and in some cases, reboot nodes.
+It may take several minutes for a configuration change to apply.
+Ensure that there are enough available nodes in your cluster to handle the evicted workload beforehand.
+
+After the configuration update is applied, all the pods in the `openshift-sriov-network-operator` namespace will change to a `Running` status.
+=====
+
+. Create the `SriovNetworkNodePolicy` object by running the following command:
++
+[source,terminal]
+----
+$ oc create -f mlx-dpdk-node-policy.yaml
+----
+
+. Create the following `SriovNetwork` object, and then save the YAML in the `mlx-dpdk-network.yaml` file.
++
+[source,yaml]
+----
+apiVersion: sriovnetwork.openshift.io/v1
+kind: SriovNetwork
+metadata:
+  name: mlx-dpdk-network
+  namespace: openshift-sriov-network-operator
+spec:
+  networkNamespace: <target_namespace>
+  ipam: |- <1>
+# ...
+  vlan: <vlan>
+  resourceName: mlxnics
+----
+<1> Specify a configuration object for the ipam CNI plug-in as a YAML block scalar. The plug-in manages IP address assignment for the attachment definition.
++
+[NOTE]
+=====
+See the "Configuring SR-IOV additional network" section for a detailed explanation on each option in the `SriovNetwork` object.
+=====
++
+An optional library, `app-netutil`, provides several API methods for gathering network information about a container's parent pod.
```

Comment:
You could reduce the number of commas in this sentence as follows:

> The `app-netutil` option library provides several API methods for gathering network information about a container's parent pod.

Reply:
You could also change "container's parent pod" to "the parent pod of the container", to avoid using a possessive:

> The app-netutil option library provides several API methods for gathering network information about the parent pod of a container.

Reply:
Applied the latter option.